crates/apub/src/objects/post.rs

```diff
@@ -108,9 +108,13 @@ impl Object for ApubPost {
   #[tracing::instrument(skip_all)]
   async fn into_json(self, context: &Data<Self::DataType>) -> LemmyResult<Page> {
     let creator_id = self.creator_id;
-    let creator = Person::read(&mut context.pool(), creator_id).await?;
+    let creator = Person::read(&mut context.pool(), creator_id)
+      .await?
+      .ok_or(LemmyErrorType::CouldntFindPerson)?;
     let community_id = self.community_id;
-    let community = Community::read(&mut context.pool(), community_id).await?;
+    let community = Community::read(&mut context.pool(), community_id)
+      .await?
+      .ok_or(LemmyErrorType::CouldntFindCommunity)?;
```

Comment:
Better to do this ok_or inside of Community::read, instead of adding it to every single callsite.

Reply:
I'd rather keep LemmyErrors and diesel errors separate, otherwise we'd have to redo all the functions in `db_schema`. And one of the main points of this was to allow call sites to do their own handling of the option returned. For 90% of cases, the `None` / missing value can be turned into a lemmy `CouldntFind` error, but some of them don't convert it to an error, and might just choose to ignore the None case.

I know its more verbose, but

Reply:
Do we really need separate errors for `CouldntFindCommunity`, `CouldntFindPost` etc? Why dont we use a single error `NotFound` instead? Then we can add a much shorter conversion method `Option.to_err()`.

Reply:
I second @Nutomic's suggestion. There are already a staggering number of error types, so I'd rather avoid adding more than is necessary.

Reply:
A single NotFound error would be a good idea if api endpoints only try to read one type of thing that could throw the error

Reply:
There were already a few `CouldntFindX` errors, probably for cases where you go to a community or person and it can't find it.

Anyway I already did the work, so I don't see the point of going from specific errors to less specific errors.